pre-commit

```diff
@@ -18,33 +16,69 @@ declare -a descriptions=(
     "AWS account number"
     "16 byte hex (e.g. S3 bucket name)"
     "NINO"
-    "IP address"
     "Private key (e.g. rsa private key, openssh private key)"
-    "Email addresses"
     "Regions embedded as part of resource descriptions"
 )
 
+declare -a email_patterns=(
+    "\b[A-Za-z0-9._%+-]{1,}@[A-Za-z0-9]{1,}.[A-Za-z0-9]{1,4}.[A-Za-z0-9]{1,4}\b"
+)
+
+declare -a email_descriptions=(
+    "Email addresses"
+)
+
+declare -a ip_patterns=(
+    "\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b"
+)
+
+declare -a ip_descriptions=(
+    "IP address"
+)
+
 if [ -d ".gitsecret" ]; then
     git secret hide
 fi
 
+
 match=0
 for i in "${!patterns[@]}"
 do
     git diff-index -p -M --cached HEAD -- | grep -v "Subproject commit" |
     grep '^+[^+]' | grep -Eq "${patterns[$i]}" &&
     echo "Blocking commit: ${descriptions[$i]} detected in patch" &&
+    echo "This part of your change triggered this issue:" &&
+    git diff-index -p -M --cached HEAD -- | grep -v "Subproject commit" |
+    grep '^+[^+]' | grep -E "${patterns[$i]}" &&
     ((match++))
 done
 
-if (( match > 0 )); then
+email_match=0
```

Comment:
I don't think you need these separate match counters, just bump `match`, it's a boolean on whether or not to allow the commit

Reply:
Possibly not, it helped when testing each for loop. :). Shall look to remove